CDCgov/prime-reportstream
# 9008 NPI sender transform

This PR adds default sender transforms to add metadata needed when sending NPI to receivers. The covid pipeline validates and adds this metadata if a valid NPI is sent. NPI validation will be added as part of ticket [9437](https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/9437) 

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Send the attached FHIR bundle to the Reports API:
[NPI test.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/11425198/NPI.test.postman_collection.txt)

2. Verify that the request was accepted

3. Open the generated HL7 file in the local sftp folder.

4. Verify ORC and OBR segments include the NPI metadata.
 
5. Run integration tests and make sure they pass
## Changes
-  Added default sender transforms for NPI
-  Fixed bug when adding extensions using sender transforms
- Added a `CA-receiver-transform.yml` to up the CA receiver to use sender trasforms

## Checklist

### Testing

### Process

## Linked Issues
- #9008 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #9437

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
